cmip6_downscaling/methods/bcsd.py

```diff
@@ -59,146 +63,231 @@ def make_flow_paths(
     tuple[str, str, str, str]
         From run hyperparameters
     """
-
     coarse_obs_path = f"{workdir}/intermediates/ERA5_{GCM}_{TRAIN_PERIOD_START}_{TRAIN_PERIOD_END}_{VARIABLE}.zarr"
     spatial_anomalies_path = f"{workdir}/intermediates/anomalies_{GCM}_{TRAIN_PERIOD_START}_{TRAIN_PERIOD_END}_{VARIABLE}.zarr"
     bias_corrected_path = f"{workdir}/intermediates/bc_{SCENARIO}_{GCM}_{TRAIN_PERIOD_START}_{TRAIN_PERIOD_END}_{VARIABLE}.zarr"
     final_out_path = f"{outdir}/bcsd_{SCENARIO}_{GCM}_{PREDICT_PERIOD_START}_{PREDICT_PERIOD_END}_{VARIABLE}.zarr"
     return coarse_obs_path, spatial_anomalies_path, bias_corrected_path, final_out_path
 
 
-def preprocess_bcsd(
-    gcm: str,
-    train_period_start: str,
-    train_period_end: str,
-    variable: str,
-    coarse_obs_path: str,
-    spatial_anomalies_path: str,
-    connection_string: str,
-    rerun: bool = True,
-) -> tuple[str, str]:
-    """Given GCM and selected obs dataset for a given variable and a given time period, at the coarse scale to match that
-    write out the coarsened version of the obs to match that GCM's grid and the spatial anomalies
-    for the obs over that time period associated with interpolating the coarse obs back to the original fine
-    obs resolution.
+def get_transformed_data():
+    ds = xr.tutorial.open_dataset('air_temperature').chunk({'lat': 50, 'lon': 50})
+    ds['air'] = ds['air'] - 273.13
+    ds['air'].attrs['units'] = 'degC'
+    # compute a dask array to confirm that dask tasks are executed on the executor's client
+    dask.array.zeros((10000, 10000), chunks=(100, 100)).mean().compute()
+    return ds
+
+
+def return_obs(train_period_start: str, train_period_end: str, variable: str) -> xarray.Dataset:
+    """Loads ERA5 observation data for given time bounds and variable
 
     Parameters
     ----------
-    gcm : str
-        Name of GCM
     train_period_start : str
-        Date for training period start (e.g. '1985')
+        Starting time bounds
     train_period_end : str
-        Date for training period end (e.g. '2015')
+        Ending time bounds
+    variable : str
+        ERA5 variable. ex: 'tasmax'
+
+    Returns
+    -------
+    xarray.Dataset
+        Loaded xarray dataset of ERA5 observation data. Chunked in time: 365
+    """
+    obs_ds = open_era5(variable, start_year=train_period_start, end_year=train_period_end)
+    obs_ds = obs_ds.chunk({'time': 365})
```

Comment:
can you add a comment about why we need to rechunk to {time: 365}? i'm wondering if i need this in other workflows